sdks/python/apache_beam/io/gcp/gcsio.py

```diff
@@ -569,11 +570,13 @@ def _gcs_object(self, path):
         bucket=bucket, object=object_path)
     return self.client.objects.Get(request)
 
-  @retry.with_exponential_backoff(
```

Comment:
Originally this will retry the whole list_prefix method. Now it will only retry the request.